src/Promact.TestCaseManagement.Core/Controllers/ModuleController.cs

```diff
@@ -0,0 +1,157 @@
+            _iProjectRepository = iProjectRepository;
+        }
+
+        #endregion
+
+        #region "Public API(s)"
+
+        /// <summary>
+        /// Method to get all modules
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <returns></returns>
+        [HttpGet("{projectId}/module")]
+        public async Task<IActionResult> GetModulesAsync(int projectId)
+        {
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+
+            return Ok(await _iModuleRepository.GetModulesAsync(projectId));
+        }
+
+        /// <summary>
+        /// Method to retrieve single module from database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <returns></returns>
+        [HttpGet("{projectId}/module/{id}")]
+        public async Task<IActionResult> GetModuleAsync(int projectId, int id)
+        {
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(module);
+        }
+
+        /// <summary>
+        /// Method to add module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPost("{projectId}/module")]
+        public async Task<IActionResult> CreateModuleAsync(int projectId, [FromBody]ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+            var module = Mapper.Map<Module>(moduleAC);
+            module.ProjectId = projectId;
+
+            return Ok(await _iModuleRepository.AddModuleAsync(module));
+        }
+
+        /// <summary>
+        /// Method to update module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPut("{projectId}/module/{id}")]
+        public async Task<IActionResult> UpdateModuleAsync(int projectId, int id, [FromBody] ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
```

Comment:
Technically It is really irrelevant to check for project. because if project does not exist the module is not going to exists either. Project is the main entity and module and scenario are child entity which can not exists without project. It is not a best practice as it really does not prevent anything